cleverhans/future/torch/attacks/carlini_wagner_l2.py

```diff
@@ -0,0 +1,205 @@
+"""The CarliniWagnerL2 attack."""
+import torch
+
+
+INF = float('inf')
```

Comment:
Looks awkward: maybe use np.inf directly in code ? (I read your thread regarding not using numpy import, but it doesn't matter since other attacks are using it and the __init_.py will lead to it being imported anyway. The only case where it would help is when someone directly imports this attack)

Reply:
"Looking awkward" is probably a subjective thing ;) Also I don't think importing all attacks would necessarily import ```numpy``` unless an attack that uses ```numpy``` has been executed.

```
Python 3.7.5 (default, Oct 25 2019, 10:52:18)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import cleverhans.future.torch.attacks as attacks
>>> np
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'np' is not defined
>>> attacks.projected_gradient_descent
<function projected_gradient_descent at 0x10eb40440>
```

But as I said in a separate thread. This is more of a coding style issue instead of anything that matters in terms of performance. I will let @npapernot or @carlini decide :)

Reply:
Fair enough (although the attached way to check if one library imports the other is incorrect)

Reply:
Ah you're right. Updated my reply above. Thanks.